src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    static final int DATE_INDEX = 0;
+    static final int NAME_INDEX = 1;
+    static final int HOUR_INDEX = 2;
+    static final int RATE_INDEX = 3;
+    private SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        dateFormat.setLenient(false);
+        try {
+            Date fromDate = dateFormat.parse(dateFrom);
+            Date toDate = dateFormat.parse(dateTo);
+
+            StringBuilder reportLines = new StringBuilder();
+            reportLines.append("Report for period ").append(dateFrom).append(" - ")
+                    .append(dateTo).append(System.lineSeparator());
+            for (int i = 0; i < names.length; i++) {
+                String name = names[i];
+                int earned = calculateEarnedSalary(name, data, fromDate, toDate, dateFormat);
+                reportLines.append(name).append(" - ")
+                        .append(earned);
+                if (i < names.length - 1) {
+                    reportLines.append(System.lineSeparator());
+                }
```

Comment:
instead of this check, just put lineSeparator at the beginning of each line (and remove it on line 22)